source/azure_iot_hub_client.c

```diff
@@ -0,0 +1,582 @@
+                                                                                   pxAzureIoTHubClient->_internal.pucAzureIoTHubClientSymmetricKey,
+                                                                                   pxAzureIoTHubClient->_internal.ulAzureIoTHubClientSymmetricKeyLength,
+                                                                                   ( uint8_t * ) xConnectInfo.pPassword, azureiotPASSWORD_MAX,
+                                                                                   &ulPasswordLength ) ) )
+        {
+            AZLogError( ( "Failed to generate auth token." ) );
+            xResult = eAzureIoTHubClientFailed;
+        }
+        else
+        {
+            xConnectInfo.cleanSession = cleanSession;
+            xConnectInfo.pClientIdentifier = ( const char * ) pxAzureIoTHubClient->_internal.pucDeviceID;
+            xConnectInfo.clientIdentifierLength = ( uint16_t ) pxAzureIoTHubClient->_internal.ulDeviceIDLength;
+            xConnectInfo.userNameLength = ( uint16_t ) xMQTTUserNameLength;
+            xConnectInfo.keepAliveSeconds = azureiothubKEEP_ALIVE_TIMEOUT_SECONDS;
+            xConnectInfo.passwordLength = ( uint16_t ) ulPasswordLength;
+
+            /* Send MQTT CONNECT packet to broker. Last Will and Testament is not used. */
+            if( ( xMQTTResult = AzureIoTMQTT_Connect( &( pxAzureIoTHubClient->_internal.xMQTTContext ),
+                                                      &xConnectInfo,
+                                                      NULL,
+                                                      ulTimeoutMilliseconds,
+                                                      &xSessionPresent ) ) != AzureIoTMQTTSuccess )
+            {
+                AZLogError( ( "Failed to establish MQTT connection: Server=%.*s, MQTTStatus=%d.",
+                              pxAzureIoTHubClient->_internal.ulHostnameLength, pxAzureIoTHubClient->_internal.pucHostname,
+                              xMQTTResult ) );
+                xResult = eAzureIoTHubClientFailed;
+            }
+            else
+            {
+                /* Successfully established a MQTT connection with the broker. */
+                AZLogInfo( ( "An MQTT connection is established with %.*s.", pxAzureIoTHubClient->_internal.ulHostnameLength,
+                             ( const char * ) pxAzureIoTHubClient->_internal.pucHostname ) );
+                xResult = eAzureIoTHubClientSuccess;
+            }
+        }
+    }
+
+    return xResult;
+}
+/*-----------------------------------------------------------*/
+
+AzureIoTHubClientResult_t AzureIoTHubClient_Disconnect( AzureIoTHubClient_t * pxAzureIoTHubClient )
+{
+    AzureIoTMQTTStatus_t xMQTTResult;
+    AzureIoTHubClientResult_t xResult;
+
+    if( pxAzureIoTHubClient == NULL )
+    {
+        AZLogError( ( "Failed to disconnect: Invalid argument." ) );
+        return eAzureIoTHubClientInvalidArgument;
```

Comment:
>return [](start = 8, length = 6)

how about like other API impl (line 442)  to assign xResult = eAzureIoTHubClientInvalidArgument; 
 and only return at end.